manta-crypto/src/ecc.rs

```diff
@@ -145,20 +145,6 @@ pub trait PreprocessedScalarMul<COM, const N: usize>: ScalarMul<COM> + Sized {
     ) -> Self::Output;
 }
 
-impl<G, COM> PreprocessedScalarMul<COM, 1> for G
-where
-    G: ScalarMul<COM>,
-{
-    #[inline]
-    fn preprocessed_scalar_mul(
-        table: &[Self; 1],
-        scalar: &Self::Scalar,
-        compiler: &mut COM,
-    ) -> Self::Output {
-        table[0].scalar_mul(scalar, compiler)
-    }
-}
```

Comment:
Because the trait should be implemented for a single value of `N`, this should not be deleted.

Reply:
Implementation of `1` conflicts with implementation of `N`. Any good way to bypass this?

Reply:
ah ok, I see the previous comment. so we are not implementing this for all `N`?

Reply:
Yes, only for the exact `N` which is correct.

Reply:
Got it thanks!

Reply:
Side note: currently we cannot put associate constant to const generic. For example, we cannot implement `ecc::PreprocessedScalarMul<Compiler<C>, Self::Scalar::MODULUS_BITS>`. We might need to implement that for every individual field.

Reply:
I have implemented this trait only for Bls12_381. I also wrote a macro to easily support other curves if we need that